modules/apps/commerce/commerce-shipping-engine-fedex/src/main/java/com/liferay/commerce/shipping/engine/fedex/internal/util/FedExCommerceShippingOptionHelper.java

```diff
@@ -676,20 +653,17 @@ private RequestedPackageLineItem _getRequestedPackageLineItem(
 		for (int i = 0; i < commerceOrderItems.size(); i++) {
 			CommerceOrderItem commerceOrderItem = commerceOrderItems.get(i);
 
-			CPInstance cpInstance = commerceOrderItem.fetchCPInstance();
-
 			Dimensions dimensions = _commerceShippingHelper.getDimensions(
-				cpInstance);
+				commerceOrderItem);
 
 			int fedExWidth = _getFedExDimension(dimensions.getWidth());
 			int fedExHeight = _getFedExDimension(dimensions.getHeight());
 			int fedExDepth = _getFedExDimension(dimensions.getDepth());
 
 			double fedExWeight = _getFedExWeight(
-				_commerceShippingHelper.getWeight(cpInstance));
+				_commerceShippingHelper.getWeight(commerceOrderItem));
 
-			BigDecimal price = _getPrice(
-				commerceOrderItem.getCPInstanceId(), 1);
+			BigDecimal price = commerceOrderItem.getUnitPrice();
```

Comment:
@luca-pellizzon if we get directly the unitPrice here we don't take into consideration promotion and discounts. I think it's more correct to take the final price of the item and divide it by the item quantity.